eli5/nn/text.py

```diff
@@ -0,0 +1,216 @@
+        # remove padding
+        pad_indices = _find_padding(pad_value=pad_value, pad_token=pad_token, doc=doc, tokens=tokens)
+        # If pad_value is not the actual padding value, behaviour is unknown
+        tokens, heatmap = _trim_padding(pad_indices, tokens, heatmap)
+    document = _construct_document(tokens)
+    spans = _build_spans(tokens, heatmap, document)
+    weighted_spans = WeightedSpans([
+        DocWeightedSpans(document, spans=spans)
+    ]) # why list? - for each vectorized - don't need multiple vectorizers?
+       # multiple highlights? - could do positive and negative expl?
+    return tokens, heatmap, weighted_spans
+
+
+def resize_1d(heatmap, length, interpolation_kind='linear'):
+    # type: (np.ndarray, int, Union[str, int]) -> np.ndarray
+    """
+    Resize the ``heatmap`` 1D array to match the specified ``length``.
+
+    For example, upscale/upsample a heatmap with length 400
+    to have length 500.
+
+    Parameters
+    ----------
+
+    heatmap : numpy.ndarray
+        Heatmap to be resized.
+
+    length : int
+        Required width.
+
+    interpolation_kind : str or int, optional
+        Interpolation method used by the underlying ``scipy.interpolate.interp1d`` resize function.
+
+        Used when resizing ``heatmap`` to the correct ``length``.
+
+        Default is ``linear``.
+
+    Returns
+    -------
+    heatmap : numpy.ndarray
+        The heatmap resized.
+    """
+    if len(heatmap.shape) == 1 and heatmap.shape[0] == 1:
+        # single weight, no batch
+        heatmap = heatmap.repeat(length)
+    else:
+        # more than length 1
+
+        # scipy.interpolate solution
+        # https://stackoverflow.com/questions/29085268/resample-a-numpy-array
+        # interp1d requires at least length 2 array
+        y = heatmap  # data to interpolate
+        x = np.linspace(0, 1, heatmap.size) # array matching y
+        interpolant = interp1d(x, y, kind=interpolation_kind) # approximates y = f(x)
+        z = np.linspace(0, 1, length)  # points where to interpolate
+        heatmap = interpolant(z)  # interpolation result
+
+        # other solutions include scipy.signal.resample (periodic, so doesn't apply)
+        # and Pillow image fromarray with mode 'F'/etc and resizing (didn't seem to work)
+    return heatmap
+
+
+def _build_spans(tokens, # type: Union[np.ndarray, list]
+                 heatmap, # type: np.ndarray
+                 document, # type: str
+                 ):
+    """Highlight ``tokens`` in ``document``, with weights from ``heatmap``."""
+    assert len(tokens) == len(heatmap)
+    spans = []
+    running = 0  # where to start looking for token in document
+    for (token, weight) in zip(tokens, heatmap):
+        # find first occurrence of token, on or after running count
+        t_start = document.index(token, running)
+        # create span
+        t_end = t_start + len(token)
+        # why start and end is list of tuples?
```

Comment:
It's useful to have a list here in case of bag of words approach, which is often used for linear models: in this case a feature is a word or n-gram (regardless of it's position in the document), and if this feature occurs in the document multiple times, then it will have several spans.

Reply:
Makes sense! I will add a comment about this in `eli5.base` if you do not mind.

Reply:
That would be great, thanks!